lib/ansible/plugins/cliconf/iosxr.py

```diff
@@ -56,56 +57,167 @@ def get_device_info(self):
 
         return device_info
 
-    def get_config(self, source='running', format='text', filter=None):
+    def configure(self, admin=False):
+        prompt = to_text(self._connection.get_prompt(), errors='surrogate_or_strict').strip()
+        if not prompt.endswith(')#'):
+            if admin and 'admin-' not in prompt:
+                self.send_command('admin')
+            self.send_command('configure terminal')
+
+    def abort(self, admin=False):
+        prompt = to_text(self._connection.get_prompt(), errors='surrogate_or_strict').strip()
+        if prompt.endswith(')#'):
+            self.send_command('abort')
+            if admin and 'admin-' in prompt:
+                self.send_command('abort')
```

Comment:
I think this should be 'exit' instead of 'abort'

Reply:
Pushing the config on device in candidate datastore and fetching the diff won't result in output. However while existing (without commit since no diff) will result in a prompt on iosxr. 
```
RP/0/0/CPU0:an-iosxr-01(config)#interface Loopback888
RP/0/0/CPU0:an-iosxr-01(config-if)#description test for ansible
RP/0/0/CPU0:an-iosxr-01(config-if)#exit
RP/0/0/CPU0:an-iosxr-01(config)#exit
Uncommitted changes found, commit them before exiting(yes/no/cancel)? [cancel]:
```
Since prompt is not handled it triggers command timeout. That's the reason abort is used

Reply:
Fixed